superset/forms.py

```diff
@@ -0,0 +1,191 @@
+"""Contains the logic to create cohesive forms on the explore view"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import logging
+from flask_babel import lazy_gettext as _
+from flask_appbuilder.fieldwidgets import BS3TextFieldWidget
+from flask_appbuilder.forms import DynamicForm
+from flask_wtf.file import FileField, FileAllowed, FileRequired
+from wtforms import (
+    BooleanField, SelectField, IntegerField, StringField)
+from wtforms.validators import DataRequired, Optional, NumberRange
+
+from superset import app, db
+import superset.models.core as models
+
+
+config = app.config
+
+
+class CsvToDatabaseForm(DynamicForm):
+    name = StringField(
+        _('Table Name'),
+        description=_('Name of table to be created from csv data.'),
+        validators=[DataRequired()],
+        widget=BS3TextFieldWidget())
+    csv_file = FileField(
+        _('CSV File'),
+        description=_('Select a CSV file to be uploaded to a database.'),
+        validators=[
+            FileRequired(), FileAllowed(['csv'], _('CSV Files Only!'))])
+    all_datasources = []
+    try:
+        all_datasources = db.session\
```

Comment:
This should be in view scope, not in module scope. As is, it would be interpreted when the web server (re)starts

Reply:
Fixed